app/gui2/src/stores/graph/graphDatabase.ts

```diff
@@ -567,6 +604,8 @@ export interface NodeDataFromAst {
   conditionalPorts: Set<Ast.AstId>
   /** An AST node containing the node's documentation comment. */
   docs: Ast.Documented | undefined
+  /** The index of the argument in the function's argument list, if the node is an input node. */
+  argIndex: number | undefined
```

Comment:
This representation seems a bit "denormalized"--almost all the other fields here are not applicable to an input node